1-Draft/RFC00XX-Update-Error-View.md

```diff
@@ -0,0 +1,181 @@
+
+- $ErrorView should contain these views
+
+    + NormalView
+    + CategoryView
+    + ConciseView
+
+- Error Message syntax for ConciseView
+
+```powershell
+ERROR: <Exception Message>: <Positional Info>
+ERROR: Cannot find path ‘C:\blah’ because it does not exist: At line:1 char:1
+```
+
+2. A new cmdlet “Resolve-ErrorRecord” will produce detailed error information similar to Format-List * -Force.
+
+- Resolve-ErrorRecord will provide the following:
+
+    + Display Last Error ($Error[0]) – default behavior
+    + Accept Pipeline input – support $error[1] |
+    + Option for the first X errors in the session
+    + Option for the Last X errors in the session
+
+- Resolve-ErrorRecord syntax
+
+```powershell
+Resolve-ErrorRecord  [-InputObject <psobject>] [-Newest <Int32>] [-First <Int32>] [-All] [<CommonParameters>]
+```
+
+First parameter set
+
+- Newest
+
+    + Datatype: int32
+    + specifies one or more of last errors to display
+    + Not required
+
+- First
+
+    + Datatype: Int32
+    + Specifies one or more of the first errors that occurred
+    + Not required
+
+- All
+
+    + Datatype: Switch
+    + Specifies display all errors in session
+    + Not required
+
+Examples
+
+```powershell
+# Displays details of the last error displayed
+Resolve-ErrorRecord
+
+# Displays error details from pipeline error object
+$error[1] | Resolve-ErrorRecord
+
+# Display detailed error information for the last 5 errors
+Resolve-ErrorRecord -Last 5
+
+# Display detailed error information for the first 3 errors
+Resolve-ErrorRecord -First 3
+
+# Display detailed information for all errors that occured in the session
+Resolve-ErrorRecord -All
+```
+__Example 1__
+
+```powershell
+PS C:\test> Get-item c:\blah
+Get-item : Cannot find path 'C:\blah' because it does not exist. At line:1 char:1
+
+PS C:\test> Resolve-ErrorRecord
+
+PSMessageDetails      :
+Exception             : System.Management.Automation.ItemNotFoundException: Cannot find path 'C:\blah' because it does
+                        not exist.
+                           at System.Management.Automation.LocationGlobber.ExpandMshGlobPath(String path, Boolean
+                        allowNonexistingPaths, PSDriveInfo drive, ContainerCmdletProvider 
+
+```
+
+__Example 2__
+
+```powershell
+PS C:\test> slkjdfh
+ERROR: The term 'slkjdfh' is not recognized as the name of a cmdlet, function, script file, or operable program. At line:1 char:1
+PS C:\test> $error[0] | Resolve-ErrorRecord
+
+PSMessageDetails      :
+Exception             : System.Management.Automation.CommandNotFoundException: The term 'slkjdfh' is not recognized as
+```
+
+## Alternate Proposals and Considerations
+
+__Alternative single-line display__
+
+```powershell
+<CommandName>:<Exception Message>
+get-item: Cannot find path ‘C:\blah’ because it does not exist:
+
```

Comment:
I would drop the position and make this the simple view. In my experience, when users see line numbers, that, coupled with red text, encourages them to go straight to the script developer.

Or better, make it conditional. Only show the position if it comes from a file. Otherwise, it is not needed in the simple view.

Reply:
Yes, don't include position (in the default output) unless it's related to an actual file

Reply:
3rded and like the conditional